oauth2/src/main/java/com/okta/spring/oauth/ConfigurableAccessTokenConverter.java

```diff
@@ -74,7 +77,16 @@ public OAuth2AccessToken extractAccessToken(String value, Map<String, ?> map) {
 
     @Override
     public OAuth2Authentication extractAuthentication(Map<String, ?> map) {
-        return super.extractAuthentication(tweakScopeMap(map));
-    }
+        OAuth2Authentication originalOAuth2Authentication = super.extractAuthentication(tweakScopeMap(map));
 
-}
+        Authentication originalUserAuthentication = originalOAuth2Authentication.getUserAuthentication();
+        if (originalUserAuthentication != null) {
+            UsernamePasswordAuthenticationToken newToken = new UsernamePasswordAuthenticationToken(originalUserAuthentication.getPrincipal(),
+                    "N/A",
+                    originalUserAuthentication.getAuthorities());
```

Comment:
With JHipster, I had to grab the "groups" claim, create authorities, and set them accordingly. 

https://github.com/jhipster/generator-jhipster/blob/master/generators/server/templates/src/main/java/package/web/rest/_AccountResource.java#L104